src/index.js

```diff
@@ -6,13 +6,24 @@ import App from "./components/app";
 import List from "./components/list";
 import Profile from "./components/profile";
 import NoMatch from "./components/no-match";
+import auth from "./util/auth";
+
+const Login = ( { location: { state } } ) => {
+	const redirectTo = state && state.nextPathname;
+	return (
+		<div>
+			<p>You must log in to use this app!</p>
+			<a href={auth.authUrl( redirectTo )}>Log In</a>
+		</div>
+	);
+};
 
 render( (
 	<Router history={ createHistory() }>
-		<Route path="/" component={ App }>
+		<Route path="/login" component={ Login } onEnter={ auth.requireAuth } />
```

Comment:
I'm ignorant. Does it means the login page requires auth?

*I'm asking for a friend.*

*my imaginary friend*